Source/WebCore/html/FormAssociatedElement.cpp

```diff
@@ -37,6 +37,12 @@ FormAssociatedElement::FormAssociatedElement(HTMLFormElement* form)
 
 HTMLFormElement* FormAssociatedElement::formForBindings() const
 {
+    if (asHTMLElement().document().settings().shadowRootReferenceTargetEnabled()) {
+        RefPtr form = this->form();
```

Comment:
Looks like this code is getting duplicated in 4-5 different places.
Can we add a helper function to share the code?